### PR TITLE
refactor: replace `once_cell` with `std::sync::OnceLock`

### PR DESCRIPTION
**Pull request by cijiugechu**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information
Since Rust 1.70, [once_cell::sync::OnceCell](https://docs.rs/once_cell/latest/once_cell/sync/struct.OnceCell.html), from the `once_cell` crate got integrated into the standard library as [std::sync::OnceLock](https://doc.rust-lang.org/std/sync/struct.OnceLock.html).

### CODE REVIEW

**Review by FabianLars — DISMISSED** [review]

File: tooling/cli/src/helpers/app_paths.rs [hunk 1]
```diff
@@ -7,16 +7,18 @@ use std::{
...
 
 use ignore::WalkBuilder;
-use once_cell::sync::Lazy;
 
 use tauri_utils::{
   config::parse::{folder_has_configuration_file, is_configuration_file, ConfigFormat},
   platform::Target,
 };
 
+static APP_DIR: OnceLock<PathBuf> = OnceLock::new();
```
Comment (amrbashir) [inline tooling/cli/src/helpers/app_paths.rs]: why was this moved? just keep it at the same place as before

File: tooling/cli/src/helpers/config.rs [hunk 2]
```diff
@@ -15,9 +14,11 @@ use std::{
   env::{current_dir, set_current_dir, set_var, var_os},
   ffi::OsStr,
   process::exit,
-  sync::{Arc, Mutex},
+  sync::{Arc, Mutex, OnceLock},
 };
 
+static CONFING_HANDLE: OnceLock<ConfigHandle> = OnceLock::new();
```
Comment (amrbashir) [inline tooling/cli/src/helpers/config.rs]: same

Comment (amrbashir) [inline tooling/cli/src/helpers/config.rs]: while you're at it, could you also fix the typo here? `CONFING` -> `CONFIG`

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for your contribution, just two small nit-picks and this should be good to go

File: tooling/cli/src/helpers/app_paths.rs [hunk 3]
```diff
@@ -114,9 +114,11 @@ fn get_app_dir() -> Option<PathBuf> {
 }
 
 pub fn app_dir() -> &'static PathBuf {
-  static APP_DIR: Lazy<PathBuf> =
-    Lazy::new(|| get_app_dir().unwrap_or_else(|| get_tauri_dir().parent().unwrap().to_path_buf()));
-  &APP_DIR
+  static APP_DIR: OnceLock<PathBuf> = OnceLock::new();
+
```
Comment (amrbashir) [inline tooling/cli/src/helpers/app_paths.rs]:
```suggestion
```

File: tooling/cli/src/helpers/config.rs [hunk 4]
```diff
@@ -115,8 +114,9 @@ pub fn nsis_settings(config: NsisConfig) -> tauri_bundler::NsisSettings {
 }
 
 fn config_handle() -> &'static ConfigHandle {
-  static CONFING_HANDLE: Lazy<ConfigHandle> = Lazy::new(Default::default);
-  &CONFING_HANDLE
+  static CONFIG_HANDLE: OnceLock<ConfigHandle> = OnceLock::new();
+
```
Comment (amrbashir) [inline tooling/cli/src/helpers/config.rs]:
```suggestion
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]
Thanks for your contributions.

**Pull request merged**